src/session_impl.cpp

```diff
@@ -174,14 +174,27 @@ namespace
 #include <winerror.h>
 #endif
 
-using libtorrent::aux::session_impl;
 using namespace std::placeholders;
 
 #ifdef BOOST_NO_EXCEPTIONS
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-noreturn"
+#endif
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wmissing-noreturn"
+#endif
 namespace boost {
-	void throw_exception(std::exception const& e) { ::abort(); }
+	void throw_exception(std::exception const&) { ::abort(); }
```

Comment:
under `BOOST_NO_EXCEPTIONS` this functions needs to be user defined, it's actually used inside `asio`. See http://www.boost.org/doc/libs/1_62_0/libs/exception/doc/throw_exception.html